examples/odd_or_even/tests/odd_or_even_tests.cpp

```diff
@@ -0,0 +1,236 @@
+}
+
+BOOST_FIXTURE_TEST_CASE(bet_and_win_one_round_mixed_fractional, odd_or_even_tester)
+{
+    const auto ses_id = new_game_session(game_name, player_name, casino_id, STRSYM("9.0000"), STRSYM("1.0000"));
+    roll(ses_id, odd_number);
+    take(ses_id);
+    check_player_win(STRSYM("4.5000"), STRSYM("0.5000"));
+}
+
+BOOST_FIXTURE_TEST_CASE(bet_and_win_one_round_mixed_larger_bonus_fractional, odd_or_even_tester)
+{
+    const auto ses_id = new_game_session(game_name, player_name, casino_id, STRSYM("1.0000"), STRSYM("9.0000"));
+    roll(ses_id, odd_number);
+    take(ses_id);
+    check_player_win(STRSYM("0.5000"), STRSYM("4.5000"));
+}
+
+
+BOOST_FIXTURE_TEST_CASE(bet_and_lose_one_round_real, odd_or_even_tester)
+{
+    const auto ses_id = new_game_session(game_name, player_name, casino_id, STRSYM("10.0000"));
+    roll(ses_id, even_number);
+    check_player_win(-STRSYM("5.0000"));
+}
+
+BOOST_FIXTURE_TEST_CASE(bet_and_lose_one_round_bonus, odd_or_even_tester)
+{
+    const auto ses_id = new_game_session(game_name, player_name, casino_id, zero_asset, STRSYM("10.0000"));
+    roll(ses_id, even_number);
+    check_player_win(zero_asset, -STRSYM("5.0000"));
+}
+
+BOOST_FIXTURE_TEST_CASE(bet_and_lose_one_round_mixed, odd_or_even_tester)
+{
+    const auto ses_id = new_game_session(game_name, player_name, casino_id, STRSYM("5.0000"), STRSYM("5.0000"));
+    roll(ses_id, even_number);
+    check_player_win(-STRSYM("2.5000"), -STRSYM("2.5000"));
+}
+
+// ============= TWO ROUNDS
+
+BOOST_FIXTURE_TEST_CASE(bet_and_win_two_rounds_real, odd_or_even_tester)
+{
+    const asset ante = STRSYM("10.0000");
+    const auto ses_id = new_game_session(game_name, player_name, casino_id, ante);
+    roll(ses_id, odd_number);
+    roll(ses_id, odd_number, ante);
+    take(ses_id);
+    check_player_win(STRSYM("10.0000"));
+}
+
+BOOST_FIXTURE_TEST_CASE(bet_and_win_two_rounds_bonus, odd_or_even_tester)
+{
+    const asset ante = STRSYM("10.0000");
+    const auto ses_id = new_game_session(game_name, player_name, casino_id, zero_asset, ante);
+    roll(ses_id, odd_number);
+    roll(ses_id, odd_number, zero_asset, ante);
+    take(ses_id);
+    check_player_win(zero_asset, STRSYM("10.0000"));
+}
+
+BOOST_FIXTURE_TEST_CASE(bet_and_win_two_rounds_mixed, odd_or_even_tester)
+{
+    const auto ses_id = new_game_session(game_name, player_name, casino_id, STRSYM("5.0000"), STRSYM("5.0000"));
+    roll(ses_id, odd_number);
+    roll(ses_id, odd_number, STRSYM("15.0000"), STRSYM("5.0000"));
+    take(ses_id);
+    check_player_win(STRSYM("10.0000"), STRSYM("5.0000"));
+}
+
+BOOST_FIXTURE_TEST_CASE(bet_and_lose_two_rounds_mixed, odd_or_even_tester)
+{
+    const auto ses_id = new_game_session(game_name, player_name, casino_id, STRSYM("5.0000"), STRSYM("5.0000"));
+    roll(ses_id, odd_number);
+    roll(ses_id, even_number, STRSYM("15.0000"), STRSYM("5.0000"));
+    check_player_win(-STRSYM("10.0000"), -STRSYM("5.0000"));
+}
+
```

Comment:
maybe should add tests where first deposit - real/mixed, second - mixed/bonus (mixed mixed),
and test with session closing

Reply:
i agree - probably worth testing real/mixed and bonus/mixed as well

mixed/mixed scenario is covered by `bet_and_win_two_rounds_mixed`

and a test with session closing is absolutely needed

Reply:
added a couple of new tests